pkg/tsdb/tempo/tempo.go

```diff
@@ -91,43 +74,68 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 	// loop over queries and execute them individually.
 	for i, q := range req.Queries {
 		ctxLogger.Debug("Processing query", "counter", i, "function", logEntrypoint())
-		if res, err := s.query(ctx, req.PluginContext, q); err != nil {
-			ctxLogger.Error("Error processing query", "error", err)
-			return response, err
-		} else {
-			if res != nil {
-				ctxLogger.Debug("Query processed", "counter", i, "function", logEntrypoint())
-				response.Responses[q.RefID] = *res
-			} else {
-				ctxLogger.Debug("Query resulted in empty response", "counter", i, "function", logEntrypoint())
+
+		var res *backend.DataResponse
+		var err error
+
+		switch q.QueryType {
+		case string(dataquery.TempoQueryTypeTraceId):
+			res, err = s.getTrace(ctx, req.PluginContext, q)
+			if err != nil {
+				ctxLogger.Error("Error processing query", "error", err)
```

Comment:
nit: after splitting we could also make error messages more specific, e.g.: `Error processing TraceId query` and `Error processing query TraceQL query`